watch.go

```diff
@@ -71,7 +85,11 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 		// if we're told to stop, close the response Body
 		go func() {
 			<-fb.stopWatching
+
+			fb.watchMtx.Lock()
 			closedManually = true
+			fb.watchMtx.Unlock()
```

Comment:
Fair enough. However, I don't think it's worth building a whole new type just for this one function. We also can't use `sync/atomic` unless we use 'unsafe.Pointer` which i'd like to stay away from.

I'll just add another mutex inside that function.